### ISSUE TITLE
`No file descriptors available` with Emscripten

### ISSUE THREAD

**Issue by mischnic** [streamich/memfs#607]
I've run into a case where a Emscripten/WASM library (libzip and/or zlib) don't accept using memfs instead of native fs because the file descriptors integers returned by `fs.open` are too large. On Linux, they default to 1-4096, and here they are much bigger

https://github.com/streamich/memfs/blob/722c5bd29d7be101151d38a3d4ce76c83e8b9dc1/src/volume.ts#L560

Starting at a small number and incrementing in `newFdNumber()` worked for me.

I don't have a need to fix this at the moment, but these findings might be interesting to someone else.

**Comment by G-Rath** [comment]
Interesting, but this strikes me as a bug in the library you're using rather than us, unless the numbers we're using are actually invalid for linux fds in which case the bug is ours.

An easy work around on our end would be to support setting that value somehow - even if via `process.env`, that'd at least unblock things.

**Comment by G-Rath** [comment]
I'm going to close this per the above, and because there doesn't seem to be a huge issue or suitable reproduction - if someone is having problems with the file descriptor integers, happy to discuss in a new issue.
